<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Exercises | Normalization (BCB420 Lecture)</title>
  <meta name="description" content="A short bookdown module introducing normalization for expression data (bulk and single-cell)." />
  <meta name="generator" content="bookdown 0.46 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Exercises | Normalization (BCB420 Lecture)" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A short bookdown module introducing normalization for expression data (bulk and single-cell)." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Exercises | Normalization (BCB420 Lecture)" />
  
  <meta name="twitter:description" content="A short bookdown module introducing normalization for expression data (bulk and single-cell)." />
  

<meta name="author" content="Ruth Isserlin" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="worked-normalization-examples.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">BCB420 Lecture 5</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a>
<ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#learning-goals"><i class="fa fa-check"></i><b>0.1</b> Learning goals</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#example-datasets"><i class="fa fa-check"></i><b>0.2</b> Example datasets</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="what-is-normalization.html"><a href="what-is-normalization.html"><i class="fa fa-check"></i><b>1</b> What is normalization?</a></li>
<li class="chapter" data-level="2" data-path="technical-vs-biological-variation.html"><a href="technical-vs-biological-variation.html"><i class="fa fa-check"></i><b>2</b> Technical vs biological variation</a>
<ul>
<li class="chapter" data-level="2.1" data-path="technical-vs-biological-variation.html"><a href="technical-vs-biological-variation.html#technical-variation"><i class="fa fa-check"></i><b>2.1</b> Technical variation</a></li>
<li class="chapter" data-level="2.2" data-path="technical-vs-biological-variation.html"><a href="technical-vs-biological-variation.html#biological-variation"><i class="fa fa-check"></i><b>2.2</b> Biological variation</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="visualizing-distributions.html"><a href="visualizing-distributions.html"><i class="fa fa-check"></i><b>3</b> Visualizing distributions</a>
<ul>
<li class="chapter" data-level="3.1" data-path="visualizing-distributions.html"><a href="visualizing-distributions.html#load-some-data"><i class="fa fa-check"></i><b>3.1</b> load Some data</a></li>
<li class="chapter" data-level="3.2" data-path="visualizing-distributions.html"><a href="visualizing-distributions.html#boxplots"><i class="fa fa-check"></i><b>3.2</b> Boxplots</a></li>
<li class="chapter" data-level="3.3" data-path="visualizing-distributions.html"><a href="visualizing-distributions.html#density-plots"><i class="fa fa-check"></i><b>3.3</b> Density plots</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="normalization-methods.html"><a href="normalization-methods.html"><i class="fa fa-check"></i><b>4</b> Normalization methods</a>
<ul>
<li class="chapter" data-level="4.1" data-path="normalization-methods.html"><a href="normalization-methods.html#library-size-scaling"><i class="fa fa-check"></i><b>4.1</b> Library-size scaling</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="normalization-methods.html"><a href="normalization-methods.html#counts-per-million-cpm"><i class="fa fa-check"></i><b>4.1.1</b> Counts-per-million (CPM)</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="normalization-methods.html"><a href="normalization-methods.html#distribution-based-normalization"><i class="fa fa-check"></i><b>4.2</b> Distribution-based normalization</a></li>
<li class="chapter" data-level="4.3" data-path="normalization-methods.html"><a href="normalization-methods.html#rna-seq-composition-aware-methods"><i class="fa fa-check"></i><b>4.3</b> RNA-seq composition-aware methods</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="normalization-methods.html"><a href="normalization-methods.html#edger-tmm"><i class="fa fa-check"></i><b>4.3.1</b> edgeR: TMM</a></li>
<li class="chapter" data-level="4.3.2" data-path="normalization-methods.html"><a href="normalization-methods.html#deseq2-size-factors"><i class="fa fa-check"></i><b>4.3.2</b> DESeq2: size factors</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="normalization-methods.html"><a href="normalization-methods.html#single-cell-note"><i class="fa fa-check"></i><b>4.4</b> Single-cell note</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="worked-normalization-examples.html"><a href="worked-normalization-examples.html"><i class="fa fa-check"></i><b>5</b> Worked normalization examples</a>
<ul>
<li class="chapter" data-level="5.1" data-path="worked-normalization-examples.html"><a href="worked-normalization-examples.html#setup-download-supplementary-files"><i class="fa fa-check"></i><b>5.1</b> Setup: download supplementary files</a></li>
<li class="chapter" data-level="5.2" data-path="worked-normalization-examples.html"><a href="worked-normalization-examples.html#load-some-data-1"><i class="fa fa-check"></i><b>5.2</b> load Some data</a></li>
<li class="chapter" data-level="5.3" data-path="worked-normalization-examples.html"><a href="worked-normalization-examples.html#example-1-gse119732"><i class="fa fa-check"></i><b>5.3</b> Example 1: GSE119732</a></li>
<li class="chapter" data-level="5.4" data-path="worked-normalization-examples.html"><a href="worked-normalization-examples.html#convert-the-raw-counts-to-counts-per-million-cpm"><i class="fa fa-check"></i><b>5.4</b> Convert the raw counts to Counts per million (CPM)</a></li>
<li class="chapter" data-level="5.5" data-path="worked-normalization-examples.html"><a href="worked-normalization-examples.html#filter-out-lowly-expressed-genes"><i class="fa fa-check"></i><b>5.5</b> filter out lowly expressed genes</a></li>
<li class="chapter" data-level="5.6" data-path="worked-normalization-examples.html"><a href="worked-normalization-examples.html#incorporate-a-design-matrix---description-of-the-samples."><i class="fa fa-check"></i><b>5.6</b> Incorporate a design matrix - description of the samples.</a></li>
<li class="chapter" data-level="5.7" data-path="worked-normalization-examples.html"><a href="worked-normalization-examples.html#normalize-dataset-using-tmm"><i class="fa fa-check"></i><b>5.7</b> Normalize Dataset using TMM</a></li>
<li class="chapter" data-level="5.8" data-path="worked-normalization-examples.html"><a href="worked-normalization-examples.html#look-at-the-distribution-of-our-samples-in-2d-space"><i class="fa fa-check"></i><b>5.8</b> Look at the distribution of our samples in 2D space</a></li>
<li class="chapter" data-level="5.9" data-path="worked-normalization-examples.html"><a href="worked-normalization-examples.html#normalize-the-dataset-using-rle"><i class="fa fa-check"></i><b>5.9</b> Normalize the Dataset using RLE</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="exercises.html"><a href="exercises.html"><i class="fa fa-check"></i><b>6</b> Exercises</a>
<ul>
<li class="chapter" data-level="6.1" data-path="exercises.html"><a href="exercises.html#exercise-1-httpswww.ncbi.nlm.nih.govgeoqueryacc.cgiaccgse119732"><i class="fa fa-check"></i><b>6.1</b> Exercise 1 — https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE119732</a></li>
<li class="chapter" data-level="6.2" data-path="exercises.html"><a href="exercises.html#exercise-2-httpswww.ncbi.nlm.nih.govgeoqueryacc.cgiaccgse122380"><i class="fa fa-check"></i><b>6.2</b> Exercise 2 — https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE122380</a></li>
<li class="chapter" data-level="6.3" data-path="exercises.html"><a href="exercises.html#exercise-3--"><i class="fa fa-check"></i><b>6.3</b> Exercise 3 -</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Normalization (BCB420 Lecture)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="exercises" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> Exercises<a href="exercises.html#exercises" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>These are intended to be done <strong>after</strong> completing the worked examples.</p>
<div id="exercise-1-httpswww.ncbi.nlm.nih.govgeoqueryacc.cgiaccgse119732" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Exercise 1 — <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE119732" class="uri">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE119732</a><a href="exercises.html#exercise-1-httpswww.ncbi.nlm.nih.govgeoqueryacc.cgiaccgse119732" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Using <strong>GSE119732</strong>,</p>
<ol style="list-style-type: decimal">
<li>Map the distribution of the samples as a box plot and a density plot using only the raw data.</li>
<li>Add colours to the plot separating the different variables in the model</li>
<li>Filter out lowly expressed genes and re plot the distributions.</li>
</ol>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="exercises.html#cb55-1" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb55-2"><a href="exercises.html#cb55-2" tabindex="-1"></a><span class="fu">library</span>(GEOquery)</span></code></pre></div>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## Loading required package: generics</code></pre>
<pre><code>## 
## Attaching package: &#39;generics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,
##     setequal, union</code></pre>
<pre><code>## 
## Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, aperm, append, as.data.frame, basename, cbind,
##     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
##     get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,
##     unsplit, which.max, which.min</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     &#39;browseVignettes()&#39;. To cite Bioconductor, see
##     &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>## Setting options(&#39;download.file.method.GEOquery&#39;=&#39;auto&#39;)</code></pre>
<pre><code>## Setting options(&#39;GEOquery.inmemory.gpl&#39;=FALSE)</code></pre>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="exercises.html#cb67-1" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb67-2"><a href="exercises.html#cb67-2" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code></pre></div>
<pre><code>## 
## Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:Biobase&#39;:
## 
##     combine</code></pre>
<pre><code>## The following objects are masked from &#39;package:BiocGenerics&#39;:
## 
##     combine, intersect, setdiff, setequal, union</code></pre>
<pre><code>## The following object is masked from &#39;package:generics&#39;:
## 
##     explain</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union</code></pre>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="exercises.html#cb74-1" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb74-2"><a href="exercises.html#cb74-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
<pre><code>## ── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
## ✔ forcats   1.0.1     ✔ purrr     1.2.0
## ✔ ggplot2   4.0.1     ✔ stringr   1.6.0
## ✔ lubridate 1.9.4     ✔ tidyr     1.3.1</code></pre>
<pre><code>## ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
## ✖ dplyr::combine()    masks Biobase::combine(), BiocGenerics::combine()
## ✖ dplyr::filter()     masks stats::filter()
## ✖ dplyr::lag()        masks stats::lag()
## ✖ ggplot2::Position() masks BiocGenerics::Position(), base::Position()
## ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="exercises.html#cb77-1" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb77-2"><a href="exercises.html#cb77-2" tabindex="-1"></a></span>
<span id="cb77-3"><a href="exercises.html#cb77-3" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;./supp_functions.R&quot;</span>) <span class="co"># Load in the required functions</span></span>
<span id="cb77-4"><a href="exercises.html#cb77-4" tabindex="-1"></a></span>
<span id="cb77-5"><a href="exercises.html#cb77-5" tabindex="-1"></a>gse <span class="ot">&lt;-</span> <span class="st">&quot;GSE119732&quot;</span></span>
<span id="cb77-6"><a href="exercises.html#cb77-6" tabindex="-1"></a>path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">&quot;data&quot;</span>, gse)</span>
<span id="cb77-7"><a href="exercises.html#cb77-7" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(path, <span class="at">pattern =</span> <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.txt.gz$|</span><span class="sc">\\</span><span class="st">.tsv.gz$|</span><span class="sc">\\</span><span class="st">.csv.gz$&quot;</span>, </span>
<span id="cb77-8"><a href="exercises.html#cb77-8" tabindex="-1"></a>                    <span class="at">full.names =</span> <span class="cn">TRUE</span>, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb77-9"><a href="exercises.html#cb77-9" tabindex="-1"></a></span>
<span id="cb77-10"><a href="exercises.html#cb77-10" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">safe_read</span>(files[<span class="dv">1</span>])</span>
<span id="cb77-11"><a href="exercises.html#cb77-11" tabindex="-1"></a></span>
<span id="cb77-12"><a href="exercises.html#cb77-12" tabindex="-1"></a><span class="fu">plot_box_1</span>(x[,<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(x)])</span></code></pre></div>
<p><img src="06-exercises_files/figure-html/unnamed-chunk-1-1.png" width="672" /></p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="exercises.html#cb78-1" tabindex="-1"></a><span class="fu">plot_density_1</span>(x[,<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(x)])</span></code></pre></div>
<p><img src="06-exercises_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="exercises.html#cb79-1" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span></code></pre></div>
<pre><code>## Loading required package: limma</code></pre>
<pre><code>## 
## Attaching package: &#39;limma&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:BiocGenerics&#39;:
## 
##     plotMA</code></pre>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="exercises.html#cb83-1" tabindex="-1"></a>x_cpm <span class="ot">&lt;-</span> <span class="fu">cpm</span>(<span class="at">y =</span> x[,<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(x)]) <span class="co"># Normalise for library size</span></span>
<span id="cb83-2"><a href="exercises.html#cb83-2" tabindex="-1"></a>to_remove <span class="ot">&lt;-</span> edgeR<span class="sc">::</span><span class="fu">filterByExpr</span>(x_cpm,<span class="at">min.count =</span> <span class="dv">3</span>) <span class="co"># Filter lowly expressed genes</span></span></code></pre></div>
<pre><code>## No group or design set. Assuming all samples belong to one group.</code></pre>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="exercises.html#cb85-1" tabindex="-1"></a>x_cpm_filtered <span class="ot">&lt;-</span> x_cpm[to_remove,]</span>
<span id="cb85-2"><a href="exercises.html#cb85-2" tabindex="-1"></a></span>
<span id="cb85-3"><a href="exercises.html#cb85-3" tabindex="-1"></a><span class="fu">plot_box_1</span>(x_cpm_filtered[,<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(x_cpm_filtered)], <span class="at">main =</span> <span class="st">&quot;GSE119732, lowly expressed genes filtered&quot;</span>)</span></code></pre></div>
<p><img src="06-exercises_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="exercises.html#cb86-1" tabindex="-1"></a><span class="fu">plot_density_1</span>(x_cpm_filtered[,<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(x_cpm_filtered)], <span class="at">main =</span> <span class="st">&quot;GSE119732, lowly expressed genes filtered&quot;</span>)</span></code></pre></div>
<p><img src="06-exercises_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
</div>
<div id="exercise-2-httpswww.ncbi.nlm.nih.govgeoqueryacc.cgiaccgse122380" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Exercise 2 — <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE122380" class="uri">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE122380</a><a href="exercises.html#exercise-2-httpswww.ncbi.nlm.nih.govgeoqueryacc.cgiaccgse122380" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Using <strong>GSE122380</strong>, confirm whether the ID column contains Ensembl IDs with version suffixes.</p>
<ol style="list-style-type: decimal">
<li>Map the distribution of the samples as a box plot and a density plot using only the raw data.</li>
<li>Add colours to the plot separating the different variables in the model</li>
<li>Filter out lowly expressed genes and re plot the distributions.</li>
</ol>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="exercises.html#cb87-1" tabindex="-1"></a>gse <span class="ot">&lt;-</span> <span class="st">&quot;GSE122380&quot;</span></span>
<span id="cb87-2"><a href="exercises.html#cb87-2" tabindex="-1"></a><span class="fu">fetch_geo_supp</span>(gse)</span></code></pre></div>
<pre><code>## Using locally cached version of supplementary file(s) GSE122380 found here:
## data/GSE122380/GSE122380_Supplementary_Data_Table_S1.xlsx</code></pre>
<pre><code>## Using locally cached version of supplementary file(s) GSE122380 found here:
## data/GSE122380/GSE122380_raw_counts.txt.gz</code></pre>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="exercises.html#cb90-1" tabindex="-1"></a>path <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">&quot;data&quot;</span>, gse)</span>
<span id="cb90-2"><a href="exercises.html#cb90-2" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(path, <span class="at">pattern =</span> <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.txt.gz$|</span><span class="sc">\\</span><span class="st">.tsv.gz$|</span><span class="sc">\\</span><span class="st">.csv.gz$&quot;</span>, </span>
<span id="cb90-3"><a href="exercises.html#cb90-3" tabindex="-1"></a>                    <span class="at">full.names =</span> <span class="cn">TRUE</span>, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb90-4"><a href="exercises.html#cb90-4" tabindex="-1"></a></span>
<span id="cb90-5"><a href="exercises.html#cb90-5" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">safe_read</span>(files[<span class="dv">1</span>])</span></code></pre></div>
<pre><code>## Warning: One or more parsing issues, call `problems()` on your data frame for details,
## e.g.:
##   dat &lt;- vroom(...)
##   problems(dat)</code></pre>
<pre><code>## Parsing issues detected in TSV — reading as space-delimited file instead.</code></pre>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="exercises.html#cb93-1" tabindex="-1"></a><span class="fu">plot_box_2</span>(x[,<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(x)])</span></code></pre></div>
<p><img src="06-exercises_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="exercises.html#cb94-1" tabindex="-1"></a><span class="fu">plot_density_2</span>(x[,<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(x)])</span></code></pre></div>
<p><img src="06-exercises_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="exercises.html#cb95-1" tabindex="-1"></a><span class="co"># Setup a design matrix</span></span>
<span id="cb95-2"><a href="exercises.html#cb95-2" tabindex="-1"></a></span>
<span id="cb95-3"><a href="exercises.html#cb95-3" tabindex="-1"></a><span class="fu">colnames</span>(x)</span></code></pre></div>
<pre><code>##   [1] &quot;Gene_id&quot;  &quot;18489_0&quot;  &quot;18489_10&quot; &quot;18489_11&quot; &quot;18489_12&quot; &quot;18489_13&quot;
##   [7] &quot;18489_14&quot; &quot;18489_15&quot; &quot;18489_1&quot;  &quot;18489_2&quot;  &quot;18489_3&quot;  &quot;18489_4&quot; 
##  [13] &quot;18489_5&quot;  &quot;18489_6&quot;  &quot;18489_7&quot;  &quot;18489_8&quot;  &quot;18489_9&quot;  &quot;18499_0&quot; 
##  [19] &quot;18499_10&quot; &quot;18499_11&quot; &quot;18499_12&quot; &quot;18499_13&quot; &quot;18499_14&quot; &quot;18499_15&quot;
##  [25] &quot;18499_1&quot;  &quot;18499_2&quot;  &quot;18499_3&quot;  &quot;18499_4&quot;  &quot;18499_5&quot;  &quot;18499_6&quot; 
##  [31] &quot;18499_7&quot;  &quot;18499_8&quot;  &quot;18499_9&quot;  &quot;18505_0&quot;  &quot;18505_10&quot; &quot;18505_11&quot;
##  [37] &quot;18505_12&quot; &quot;18505_13&quot; &quot;18505_14&quot; &quot;18505_15&quot; &quot;18505_1&quot;  &quot;18505_2&quot; 
##  [43] &quot;18505_3&quot;  &quot;18505_4&quot;  &quot;18505_5&quot;  &quot;18505_6&quot;  &quot;18505_7&quot;  &quot;18505_8&quot; 
##  [49] &quot;18505_9&quot;  &quot;18508_0&quot;  &quot;18508_10&quot; &quot;18508_11&quot; &quot;18508_12&quot; &quot;18508_13&quot;
##  [55] &quot;18508_14&quot; &quot;18508_15&quot; &quot;18508_1&quot;  &quot;18508_2&quot;  &quot;18508_3&quot;  &quot;18508_4&quot; 
##  [61] &quot;18508_5&quot;  &quot;18508_6&quot;  &quot;18508_7&quot;  &quot;18508_8&quot;  &quot;18508_9&quot;  &quot;18511_0&quot; 
##  [67] &quot;18511_10&quot; &quot;18511_11&quot; &quot;18511_12&quot; &quot;18511_13&quot; &quot;18511_14&quot; &quot;18511_15&quot;
##  [73] &quot;18511_1&quot;  &quot;18511_2&quot;  &quot;18511_3&quot;  &quot;18511_4&quot;  &quot;18511_5&quot;  &quot;18511_6&quot; 
##  [79] &quot;18511_7&quot;  &quot;18511_8&quot;  &quot;18511_9&quot;  &quot;18517_0&quot;  &quot;18517_10&quot; &quot;18517_11&quot;
##  [85] &quot;18517_12&quot; &quot;18517_13&quot; &quot;18517_14&quot; &quot;18517_15&quot; &quot;18517_1&quot;  &quot;18517_2&quot; 
##  [91] &quot;18517_3&quot;  &quot;18517_4&quot;  &quot;18517_5&quot;  &quot;18517_6&quot;  &quot;18517_7&quot;  &quot;18517_8&quot; 
##  [97] &quot;18517_9&quot;  &quot;18520_0&quot;  &quot;18520_10&quot; &quot;18520_11&quot; &quot;18520_12&quot; &quot;18520_13&quot;
## [103] &quot;18520_14&quot; &quot;18520_15&quot; &quot;18520_1&quot;  &quot;18520_3&quot;  &quot;18520_4&quot;  &quot;18520_5&quot; 
## [109] &quot;18520_6&quot;  &quot;18520_7&quot;  &quot;18520_8&quot;  &quot;18520_9&quot;  &quot;18855_0&quot;  &quot;18855_10&quot;
## [115] &quot;18855_11&quot; &quot;18855_12&quot; &quot;18855_13&quot; &quot;18855_14&quot; &quot;18855_15&quot; &quot;18855_1&quot; 
## [121] &quot;18855_3&quot;  &quot;18855_4&quot;  &quot;18855_5&quot;  &quot;18855_6&quot;  &quot;18855_7&quot;  &quot;18855_8&quot; 
## [127] &quot;18855_9&quot;  &quot;18858_0&quot;  &quot;18858_10&quot; &quot;18858_11&quot; &quot;18858_12&quot; &quot;18858_13&quot;
## [133] &quot;18858_14&quot; &quot;18858_15&quot; &quot;18858_1&quot;  &quot;18858_2&quot;  &quot;18858_3&quot;  &quot;18858_4&quot; 
## [139] &quot;18858_5&quot;  &quot;18858_6&quot;  &quot;18858_7&quot;  &quot;18858_8&quot;  &quot;18858_9&quot;  &quot;18870_0&quot; 
## [145] &quot;18870_10&quot; &quot;18870_11&quot; &quot;18870_12&quot; &quot;18870_13&quot; &quot;18870_14&quot; &quot;18870_15&quot;
## [151] &quot;18870_1&quot;  &quot;18870_2&quot;  &quot;18870_3&quot;  &quot;18870_5&quot;  &quot;18870_6&quot;  &quot;18870_7&quot; 
## [157] &quot;18870_8&quot;  &quot;18870_9&quot;  &quot;18907_0&quot;  &quot;18907_10&quot; &quot;18907_11&quot; &quot;18907_12&quot;
## [163] &quot;18907_13&quot; &quot;18907_14&quot; &quot;18907_15&quot; &quot;18907_1&quot;  &quot;18907_2&quot;  &quot;18907_3&quot; 
## [169] &quot;18907_4&quot;  &quot;18907_5&quot;  &quot;18907_6&quot;  &quot;18907_7&quot;  &quot;18907_8&quot;  &quot;18907_9&quot; 
## [175] &quot;18912_0&quot;  &quot;18912_10&quot; &quot;18912_11&quot; &quot;18912_12&quot; &quot;18912_13&quot; &quot;18912_14&quot;
## [181] &quot;18912_15&quot; &quot;18912_1&quot;  &quot;18912_2&quot;  &quot;18912_3&quot;  &quot;18912_4&quot;  &quot;18912_5&quot; 
## [187] &quot;18912_6&quot;  &quot;18912_7&quot;  &quot;18912_8&quot;  &quot;18912_9&quot;  &quot;19093_0&quot;  &quot;19093_10&quot;
## [193] &quot;19093_11&quot; &quot;19093_12&quot; &quot;19093_13&quot; &quot;19093_14&quot; &quot;19093_15&quot; &quot;19093_1&quot; 
## [199] &quot;19093_2&quot;  &quot;19093_3&quot;  &quot;19093_4&quot;  &quot;19093_5&quot;  &quot;19093_6&quot;  &quot;19093_7&quot; 
## [205] &quot;19093_8&quot;  &quot;19093_9&quot;  &quot;19108_0&quot;  &quot;19108_10&quot; &quot;19108_11&quot; &quot;19108_12&quot;
## [211] &quot;19108_13&quot; &quot;19108_14&quot; &quot;19108_15&quot; &quot;19108_1&quot;  &quot;19108_2&quot;  &quot;19108_3&quot; 
## [217] &quot;19108_5&quot;  &quot;19108_6&quot;  &quot;19108_7&quot;  &quot;19108_8&quot;  &quot;19108_9&quot;  &quot;19127_0&quot; 
## [223] &quot;19127_10&quot; &quot;19127_11&quot; &quot;19127_12&quot; &quot;19127_13&quot; &quot;19127_14&quot; &quot;19127_15&quot;
## [229] &quot;19127_1&quot;  &quot;19127_2&quot;  &quot;19127_3&quot;  &quot;19127_4&quot;  &quot;19127_5&quot;  &quot;19127_6&quot; 
## [235] &quot;19127_7&quot;  &quot;19127_8&quot;  &quot;19127_9&quot;  &quot;19159_0&quot;  &quot;19159_10&quot; &quot;19159_11&quot;
## [241] &quot;19159_12&quot; &quot;19159_13&quot; &quot;19159_14&quot; &quot;19159_15&quot; &quot;19159_1&quot;  &quot;19159_2&quot; 
## [247] &quot;19159_3&quot;  &quot;19159_4&quot;  &quot;19159_5&quot;  &quot;19159_6&quot;  &quot;19159_7&quot;  &quot;19159_8&quot; 
## [253] &quot;19159_9&quot;  &quot;19190_0&quot;  &quot;19190_10&quot; &quot;19190_11&quot; &quot;19190_12&quot; &quot;19190_13&quot;
## [259] &quot;19190_14&quot; &quot;19190_15&quot; &quot;19190_1&quot;  &quot;19190_2&quot;  &quot;19190_3&quot;  &quot;19190_4&quot; 
## [265] &quot;19190_5&quot;  &quot;19190_6&quot;  &quot;19190_7&quot;  &quot;19190_8&quot;  &quot;19190_9&quot;  &quot;19193_0&quot; 
## [271] &quot;19193_10&quot; &quot;19193_11&quot; &quot;19193_12&quot; &quot;19193_13&quot; &quot;19193_14&quot; &quot;19193_15&quot;
## [277] &quot;19193_1&quot;  &quot;19193_2&quot;  &quot;19193_3&quot;  &quot;19193_5&quot;  &quot;19193_6&quot;  &quot;19193_7&quot; 
## [283] &quot;19193_8&quot;  &quot;19193_9&quot;  &quot;19209_0&quot;  &quot;19209_10&quot; &quot;19209_11&quot; &quot;19209_12&quot;
## [289] &quot;19209_14&quot; &quot;19209_15&quot; &quot;19209_1&quot;  &quot;19209_3&quot;  &quot;19209_4&quot;  &quot;19209_5&quot; 
## [295] &quot;19209_6&quot;  &quot;19209_7&quot;  &quot;19209_8&quot;  &quot;19209_9&quot;</code></pre>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="exercises.html#cb97-1" tabindex="-1"></a>samples <span class="ot">&lt;-</span> <span class="fu">colnames</span>(x)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(x)]</span>
<span id="cb97-2"><a href="exercises.html#cb97-2" tabindex="-1"></a>cell_line <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">lapply</span>(samples,<span class="at">FUN =</span> <span class="cf">function</span>(x){<span class="fu">unlist</span>(<span class="fu">strsplit</span>(x,<span class="at">split =</span> <span class="st">&quot;_&quot;</span>))[<span class="dv">1</span>]}))</span>
<span id="cb97-3"><a href="exercises.html#cb97-3" tabindex="-1"></a>time <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">lapply</span>(samples,<span class="at">FUN =</span> <span class="cf">function</span>(x){<span class="fu">unlist</span>(<span class="fu">strsplit</span>(x,<span class="at">split =</span> <span class="st">&quot;_&quot;</span>))[<span class="dv">2</span>]}))</span>
<span id="cb97-4"><a href="exercises.html#cb97-4" tabindex="-1"></a></span>
<span id="cb97-5"><a href="exercises.html#cb97-5" tabindex="-1"></a>sample_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(samples, cell_line, time)</span>
<span id="cb97-6"><a href="exercises.html#cb97-6" tabindex="-1"></a></span>
<span id="cb97-7"><a href="exercises.html#cb97-7" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> time, <span class="at">data =</span> sample_data)</span>
<span id="cb97-8"><a href="exercises.html#cb97-8" tabindex="-1"></a><span class="fu">rownames</span>(design) <span class="ot">&lt;-</span> sample_data<span class="sc">$</span>samples</span>
<span id="cb97-9"><a href="exercises.html#cb97-9" tabindex="-1"></a><span class="fu">colnames</span>(design) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;time&quot;</span>, <span class="fu">levels</span>(<span class="fu">factor</span>(time)))</span>
<span id="cb97-10"><a href="exercises.html#cb97-10" tabindex="-1"></a></span>
<span id="cb97-11"><a href="exercises.html#cb97-11" tabindex="-1"></a>x_cpm <span class="ot">&lt;-</span> <span class="fu">cpm</span>(<span class="at">y =</span> x[,<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(x)])</span>
<span id="cb97-12"><a href="exercises.html#cb97-12" tabindex="-1"></a>to_remove_withdesign <span class="ot">&lt;-</span> <span class="fu">filterByExpr</span>(x_cpm, <span class="at">min.count =</span> <span class="dv">3</span>, <span class="at">design =</span> design)</span>
<span id="cb97-13"><a href="exercises.html#cb97-13" tabindex="-1"></a>x_cpm_filtered_withdesign <span class="ot">&lt;-</span> x_cpm[to_remove_withdesign,]</span>
<span id="cb97-14"><a href="exercises.html#cb97-14" tabindex="-1"></a></span>
<span id="cb97-15"><a href="exercises.html#cb97-15" tabindex="-1"></a><span class="fu">colnames</span>(x_cpm_filtered_withdesign) <span class="ot">&lt;-</span> <span class="fu">paste</span>(sample_data<span class="sc">$</span>time,</span>
<span id="cb97-16"><a href="exercises.html#cb97-16" tabindex="-1"></a>                                             <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(sample_data), <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb97-17"><a href="exercises.html#cb97-17" tabindex="-1"></a></span>
<span id="cb97-18"><a href="exercises.html#cb97-18" tabindex="-1"></a><span class="fu">plot_box_2</span>(x_cpm_filtered_withdesign,</span>
<span id="cb97-19"><a href="exercises.html#cb97-19" tabindex="-1"></a>           <span class="at">main =</span> <span class="st">&quot;GSE122380, lowly expressed genes filtered with design&quot;</span>)</span></code></pre></div>
<p><img src="06-exercises_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="exercises.html#cb98-1" tabindex="-1"></a><span class="fu">plot_density_2</span>(x_cpm_filtered_withdesign,</span>
<span id="cb98-2"><a href="exercises.html#cb98-2" tabindex="-1"></a>           <span class="at">main =</span> <span class="st">&quot;GSE122380, lowly expressed genes filtered with design&quot;</span>)</span></code></pre></div>
<p><img src="06-exercises_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
</div>
<div id="exercise-3--" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> Exercise 3 -<a href="exercises.html#exercise-3--" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Can you use the worked example to process the above two GEO records? How?</p>
<p>Yes, but we need to change things that are specific to each GEO record, such as the design matrix. Supplementary functions also are changed by adding new functions to compensate for the way different records organise different samples and variables.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="worked-normalization-examples.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": null,
    "text": null
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["bcb420_normalization.pdf", "bcb420_normalization.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section"
  }
});
});
</script>

</body>

</html>
